src/debug_menu.cpp

```diff
@@ -380,7 +380,7 @@ void character_edit_menu()
     }
     const size_t index = charmenu.ret;
     // The NPC is also required for "Add mission", so has to be in this scope
-    npc *np = g->critter_at<npc>( locations[index] );
+    npc *np = g->critter_at<npc>( locations[index], false, true );
```

Comment:
There should now be a wrapper function (e.g. something like `ridding_critter_at`), because nobody is going to recognize what those two boolean values means here. (Even the first boolean is one two many.)

Reply:
I guess itd have to be a more general player_or_npc_at() function, because critter_at() can no longer rely on the fact that only one critter is at one position. all places that expect a player or npc from critter_at() can no longer trust they wont get a nullptr from critter_at() finding the monster there first.

Reply:
You can change the implementations of `critter_at<player>` and `critter_at<npc>` and even for `Character` to check for non-monsters first.

Reply:
Ok well... I did ... something, it doesnt feel right, but it works. 
in critter_at() I added a check for what `T` was, and only returned a monster, if the monster was not being ridden, or if `T` was specifically looking for a `monster` or `Creature`
Hope thats ok?